src/pyhf/tensor/numpy_backend.py

```diff
@@ -38,6 +38,15 @@ class numpy_backend(object):
 
     def __init__(self, **kwargs):
         self.name = 'numpy'
+        self._mode = 'CPU'
+
+    @property
+    def mode(self):
+        return self._mode
+
+    @mode.setter
```

Comment:
To make this easier to deal with -- we shouldn't allow the mode to be changed after initialization. If the user wants to change the mode, they should supply a new backend...